lib/SILOptimizer/Transforms/PerformanceInliner.cpp

```diff
@@ -909,6 +914,17 @@ bool SILPerformanceInliner::inlineCallsIntoFunction(SILFunction *Caller) {
   if (AppliesToInline.empty())
     return false;
 
+  // If our Caller has ownership and any of our callees do not have ownership,
+  // do not inline. We will perform inlining later once our caller has had its
+  // ownership lowered.
```

Comment:
It doesn't make sense to me to bail out. Inlining a call-site should not be affected by the presence of an unrelated call site. Violating unwritten contracts causes instability and confusion.

Reply:
I found that such a condition was destabilizing the perf inliner. That being said, if they are always inline, maybe it doesn't matter.

Reply:
> I found that such a condition was destabilizing the perf inliner. That being said, if they are always inline, maybe it doesn't matter.

Sorry, I don't get it. Certainly in theory, inlining decisions should not be significantly affected by other unrelated calls that are not inlined.

Reply:
I found that when I added an extra run of the early inliner that the inlining decisions would differ since we would inline all of the ossa and then later the non-ossa code. But this is just for always inline at this point, so I don't think what I saw before will happen.

Reply:
I put it back to how it was before.